test/NinjaTools.FluentMockServer.API.Tests/Downstream/RequestMapperSpec.cs

```diff
@@ -0,0 +1,355 @@
+        {
+            this.Given(_ => GivenInputRequestHasValidRequestUri("http://google.de"))
+                .And(_ => GivenTheInputRequestHasMethod(request.Method))
+                .When(_ => WhenMapped())
+                .Then(_ => ThenNoErrorIsReturned())
+                .Then(_ => ThenTheMappedRequestHasMethod(expected.Method))
+                .BDDfy();
+        }
+
+
+
+        [BddfyTheory]
+        [ClassData(typeof(HttpRequestMapperData))]
+        public void MapAsync_Does_Not_Map_Null_Content(HttpRequest request, HttpRequestMessage expected)
+        {
+            this.Given(_ => GivenInputRequestHasValidRequestUri(expected.RequestUri.ToString()))
+                .And(_ => GivenTheInputRequestHasMethod(request.Method))
+                .And(_ => GivenTheInputRequestHasNullContent())
+                .When(_ => WhenMapped())
+                .Then(_ => ThenNoErrorIsReturned())
+                .And(_ => ThenTheMappedContentIsNull())
+                .BDDfy();
+        }
+
+
+        [BddfyFact]
+        public void MapAsync_Does_Not_Map_HostHeader()
+        {
+            this.Given(_ => GivenInputRequestHasValidRequestUri("https://google.net:443/some/path?query=30"))
+                .And(_ => GivenTheInputRequestHasMethod("GET"))
+                .When(_ => WhenMapped())
+                .Then(_ => ThenNoErrorIsReturned())
+                .And(_ => ThenHostHeaderIsNotMapped())
+                .BDDfy();
+        }
+        
+        [BddfyTheory]
+        [ClassData(typeof(HttpContentEmailData))]
+        public void MapAsync_Maps_HttpContent(Stream body, HttpContent expected)
+        {
+            this.Given(_ => GivenInputRequestHasValidRequestUri("http://example.org"))
+                .And(_ => GivenTheInputRequestHasMethod(HttpMethod.Post.ToString()))
+                .And(_ => GivenTheInputRequestHasBody(body))
+                .When(_ => WhenMapped())
+                .And(_ => ThenNoErrorIsReturned())
+                .Then(_ => ThenTheMappedContentIs(expected))
+                .BDDfy("mapped HttpContent.");
+        }
+
+
+        private void ThenTheMappedHasRequestUri(Uri requestUri)
+        {
+            _result.Data?.RequestUri.ShouldBe(requestUri);
+        }
+
+
+        private void ThenTheMappedContentIsNull()
+        {
+            _result.Data?.Content.ShouldBeNull();
+        }
+
+
+        private void ThenHostHeaderIsNotMapped()
+        {
+            _result.Data.Headers.Host.ShouldBeNull();
+        }
+
+      
+        private void GivenTheInputRequestHasMethod([NotNull] string method)
+        {
+            _inputRequest.Method = new HttpMethod(method).Method;
+        }
+
+        private void GivenTheInputRequestHasHostWithPort(string host, int? port = null)
+        {
+            _inputRequest.Host = port.HasValue
+                ? new HostString(host, port.Value)
+                : new HostString(host);
+        }
+
+
+        private void GivenTheInputRequestHasNoHeaders()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'GivenTheInputRequestHasNoHeaders'.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)